csrc/multidevice/communication.h

```diff
@@ -15,43 +15,39 @@
 
 namespace nvfuser {
 
-/*
-  This struct gathers all the parameters necessary for the
-  construction a communication
-*/
+// This struct gathers all the parameters necessary for the
+// construction a communication
 struct CommParams {
   DeviceIdxType root = -1;
-  std::vector<at::Tensor> src_bufs;
-  std::vector<at::Tensor> dst_bufs;
-  Team team; // should not have duplicate
+  bool is_root_in_mesh = true;
```

Comment:
this shouldn't be a param's field that can be set, this field is entirely deduced from `root` and `team` members. There shouldn't be such a member in the struct

Reply:
>  this field is entirely deduced from root and team members

`team` always contains `root` IIUC. Do you mean to let params store root and **mesh** instead?

Reply:
I'll think about it. One benefit of separating compilation from execution (and thus PR) is to avoid doing too much at runFusion time, e.g., having to analyze TVs and extract information needed for execution.

Reply:
> I'll think about it. One benefit of separating compilation from execution (and thus PR) is to avoid doing too much at runFusion time, e.g., having to analyze TVs and extract information needed for execution.

Agreed. And let's stick with that for this pr!

On the other hand (for the future) I also see motivation for storing the TV and Expr ptrs:
- it contains the full info 1) which is useful for printing e.g. IO tensors, and, 2) it can be needed in the future when the lowering becomes more fleshed up
- it is more clean/concise/structured way to encode the symbolic Communication

What we could do to achieve best of both world is to store (non-mutable) pointers to IO TensorViews and Expr and precompute as class private data whats needed at instantiation.

Reply:
With #2185 , `Communication` will be an `Expr` which automatically contains input and output TensorView. That should address your concern?

Reply:
I gave up on the team=>mesh change for this PR. It would cause even more divergence with #2185 , so we'll do that in a separate PR if needed.

Reply:
> With #2185 , `Communication` will be an `Expr` which automatically contains input and output TensorView. That should address your concern?

Yes! That would be a good way of doing it. If we follow this path, I think we can completely get rid of the struct `CommunicationParams` (we then only need to pass to the constructor the ReductionOpType, or a pointer to the set/reduction Expr*)